base/broadcast.jl

```diff
@@ -39,16 +52,27 @@ promote_containertype(::Type{Array}, ct) = Array
 promote_containertype(ct, ::Type{Array}) = Array
 promote_containertype(::Type{Tuple}, ::Type{Any}) = Tuple
 promote_containertype(::Type{Any}, ::Type{Tuple}) = Tuple
+promote_containertype(::Type{Any}, ::Type{Nullable}) = Nullable
+promote_containertype(::Type{Nullable}, ::Type{Any}) = Nullable
+promote_containertype(::Type{Nullable}, ::Type{Array}) = Array{Nullable}
+promote_containertype(::Type{Array}, ::Type{Nullable}) = Array{Nullable}
+promote_containertype(::Type{Array{Nullable}}, ::Type{Array{Nullable}}) =
+    Array{Nullable}
+promote_containertype(::Type{Array{Nullable}}, ::Type{Array}) = Array{Nullable}
+promote_containertype(::Type{Array}, ::Type{Array{Nullable}}) = Array{Nullable}
+promote_containertype(::Type{Array{Nullable}}, ct) = Array{Nullable}
+promote_containertype(ct, ::Type{Array{Nullable}}) = Array{Nullable}
 promote_containertype{T}(::Type{T}, ::Type{T}) = T
 
 ## Calculate the broadcast indices of the arguments, or error if incompatible
 # array inputs
 broadcast_indices() = ()
 broadcast_indices(A) = broadcast_indices(containertype(A), A)
-broadcast_indices(::Type{Any}, A) = ()
+broadcast_indices(::Union{Type{Any}, Type{Nullable}}, A) = ()
 broadcast_indices(::Type{Tuple}, A) = (OneTo(length(A)),)
 broadcast_indices(::Type{Array}, A) = indices(A)
 broadcast_indices(::Type{Array}, A::Ref) = ()
+broadcast_indices(::Union{Type{Array}, Type{Array{Nullable}}}, A) = indices(A)
```

Comment:
Remove this one and the `broadcast_indices(::Type{Array}, A) = indices(A)` two lines above and use `broadcast_indices{T<:Array}(::Type{T}, A) = indices(A)` instead